js/check.js

```diff
@@ -0,0 +1,47 @@
+import showError from "./showError.js";
+
+export const checkEmpty = (tag) => {
+  if(tag.value === '') {
+    if(tag.id === 'email'){
+      showError(tag, '이메일을 입력해주세요.');
+      return false;
+    }
+    if(tag.id === 'password') {
+      showError(tag, '비밀번호를 입력해주세요.');
+      return false;
+    }
+    if(tag.id === 'password-check') {
+      showError(tag, '비밀번호를 입력해주세요.');
+      return false;
+    }
+  }
+  return true;
+}
+
+export const checkEmail = (tag) => {
+  if(tag.value === '') return;
+  const emailValue = tag.value.trim();
+  const emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+  if (emailRegex.test(emailValue)) {
+    if(tag.parentNode.lastChild.className ==='error-message') {
+      tag.parentNode.removeChild(tag.parentNode.lastChild);
+      tag.classList.remove('input--error')
+      return true;
+    }
+    else return true;
+  }
+  else {
+    showError(tag, '올바른 이메일 주소가 아닙니다.');
+    return false;
+  }
+}
+
+export const checkPassword = (tag) => {
+  const passwordRegex = /^(?=.*[A-Za-z])(?=.*\d).{8,}$/;
+  if (passwordRegex.test(tag.value)) return true;
+  else {
+    if(tag.value === '') return false;
+    showError(tag, '비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.');
+    return false;
+  }
```

Comment:
👍  passwordRegex을 사용하는 곳에서만 선언해서 사용하는 군요. 41L에서 return 해서 함수를 빨리 끝내는 것도 좋습니다.

return은 함수의 호출을 끝내기 때문에, else를 제거해서 같은 동작으로 바꿀 수 있습니다. 
이런 표현/테크닉을 early return 이라고 합니다.

```suggestion
  const passwordRegex = /^(?=.*[A-Za-z])(?=.*\d).{8,}$/;
  if (passwordRegex.test(tag.value)) return true;
  if(tag.value === '') return false;
  
  showError(tag, '비밀번호는 영문, 숫자 조합 8자 이상 입력해 주세요.');
  return false;
```